### PR TITLE
Fixes email forwarding link

### PR DESCRIPTION
**Pull request by belcherj**
#### Changes proposed in this Pull Request

* Adds logic to determine if a site can use email forwarding
* Adds new lib functions to decide what screens to show on email page
* Adds tests for new lib functions

#### Testing instructions

* Have site mapped to wpcom
* Have site registered at wpcom but mapped out
* Have site registered at wpcom

* goto sites
* goto domains
* click email tab
* Observe
* Click email forwarding if avail
* Go back
* click on domain then email
* Click email forwarding if avail

If a domain is mapped to wpcom and does not have G Suite you should see email forwarding option
If a domain is registered at wpcom and does not have G Suite you should see email forwarding option
If a domain is registered at wpcom and pointed elsewhere you should not see email forwarding option
If a domain already has G Suite you should not see email forwarding option

### CODE REVIEW

File: client/lib/domains/email-forwarding/test/index.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+				name: 'foo.blog',
+				type: 'SITE_REDIRECT',
+				googleAppsSubscription: { status: 'no_subscription' },
+			};
+			expect( getEmailForwardingSupportedDomains( [ siteRedirect ] ) ).toEqual( [] );
+		} );
+	} );
+} );
+
+// { googleAppsSubscription: { status: 'no_subscription' } }
```
Comment (aidvu) [inline client/lib/domains/email-forwarding/test/index.js]: Prolly just forgot.

File: client/lib/domains/email-forwarding/index.js [hunk 2]
```diff
@@ -1,11 +1,49 @@
...
+ *
+ * @param {Array} domains - list of domain objects
+ * @returns {Array} - Array of Email Forwarding supported domans
+ */
+function getEmailForwardingSupportedDomains( domains ) {
+	return domains.filter( function( domain ) {
+		const domainHasGSuite = hasGSuite( domain );
+		const wpcomHosted =
+			includes( [ domainTypes.REGISTERED ], domain.type ) && domain.hasWpcomNameservers;
+		const mapped = includes( [ domainTypes.MAPPED ], domain.type );
```
Comment (aidvu) [inline client/lib/domains/email-forwarding/index.js]: This one also needs WP.com nameservers, I believe, so we might make it simpler if the above includes handled both mapping/registration.

File: client/lib/domains/email-forwarding/test/index.js (hunk 1 above)
Comment (belcherj) [reply client/lib/domains/email-forwarding/test/index.js]: Oops, fixed

File: client/lib/domains/email-forwarding/index.js (hunk 2 above)
Comment (belcherj) [reply client/lib/domains/email-forwarding/index.js]: For MAPPED domains the value hasWpcomNameservers is undefined.  Got hit by that bug over on the G Suite functions a while back.

**Pull request merged**